node/actors/network/src/gossip/batch_signatures.rs

```diff
@@ -0,0 +1,102 @@
+//! Global state distributed by active attesters, observed by all the nodes in the network.
+use crate::watch::Watch;
+use std::{collections::HashSet, sync::Arc};
+use zksync_concurrency::sync;
+use zksync_consensus_roles::attester::{self, L1Batch};
+
+/// Mapping from attester::PublicKey to a signed attester::L1Batch message.
+/// Represents the currents state of node's knowledge about the attester signatures.
+#[derive(Clone, Default, PartialEq, Eq)]
+pub(crate) struct L1BatchSignatures(
+    pub(super) im::HashMap<attester::PublicKey, Arc<attester::SignedBatchMsg<L1Batch>>>,
```

Comment:
We need to be a bit more subtle with batch signatures retention than with network addresses: node should keep collecting and broadcasting signatures for batch N until it has a certificate for batch N (then it can just broadcast the whole certificate until the certificate is sent to L1). It shouldn't discard signatures for batch N just because an attester has sent a signature for batch N+1. IMO before we implement this broadcasting logic, we should design the API through which the node will receive the the batches to sign (from the business logic) and send back the commitqcs collected, so that we know exactly what properties the batch broadcasting should have.

Reply:
I propose [this interface](https://www.notion.so/matterlabs/L1-batches-and-Consensus-7348aef31a514940b7779e2293ba0da0?pvs=4#52f1bd16fe2f4588b88bc36ea16d0db4). @pompon0 @IAvecilla PTAL

Reply:
Seems good for now. We'll probably need a way to block while waiting for new batches, as you already mentioned. The State Keeper struct was added in the network actor; in the future it probably would be better to have it in a new actor that takes care of the related work with L1 batches. LMK what you think

Reply:
Let's name it PersistentBatchStore and move it to storage crate. I suppose that it should be embedded somehow into the BlockStore, but I'll think about it more.